src/ansys/fluent/core/services/field_data.py

```diff
@@ -638,7 +638,56 @@ class PathlinesFieldDataRequest(NamedTuple):
     zones: list | None = None
 
 
-class TFieldData:
+class _IFieldData:
+    """The base field data interface."""
+
+    def __init__(self):
+        """__init__ method of _IFieldData class."""
+        self.data = None
+        self._field_info = None
+        self._allowed_surface_names = None
+
+    def get_surface_ids(self, surfaces: List[str | int]) -> List[int]:
+        """Get a list of surface ids based on surfaces provided as inputs."""
+        return _get_surface_ids(
+            field_info=self._field_info,
+            allowed_surface_names=self._allowed_surface_names,
+            surfaces=surfaces,
+        )
+
+    def _get_scalar_field_data(self, **kwargs):
+        pass
+
+    def _get_surface_data(self, **kwargs):
+        pass
+
+    def _get_vector_field_data(self, **kwargs):
+        pass
+
+    def _get_pathlines_field_data(self, **kwargs):
+        pass
+
+    def get_field_data(
+        self,
+        obj: [
+            SurfaceFieldDataRequest
+            | ScalarFieldDataRequest
+            | VectorFieldDataRequest
+            | PathlinesFieldDataRequest
+        ],
```

Comment:
@prmukherj Annotation is incorrect (`[...]`).

Reply:
Thanks, Done.